app/templates/carpools/show.html

```diff
@@ -39,6 +39,9 @@ <h4>Carpool Status</h4>
                         <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
                         <button class="left-button carpool-control-button" type="submit" name="cancel_button">Cancel your seat in carpool</button>
                     </form>
+                    {% if current_user.has_roles('admin') %}
```

Comment:
this code is repeated three times. Maybe have at the bottom, once?

Reply:
I don't see any way to move it down without keeping the opening and closing tags for the DIV elements together. Right now every role-based branch contains the two level DIVs. 

The only other way I see is to put this upside down and put on the top level the DIVs and for each DIVs decide what to present based on the role. 

Is that what you are suggesting or am I missing some simple solution there?